schema.proto

```diff
@@ -40,6 +46,21 @@ message Block {
   repeated FactV2 facts_v2 = 4;
   repeated RuleV2 rules_v2 = 5;
   repeated CheckV2 checks_v2 = 6;
+  repeated Scope scope = 7;
```

Comment:
this allows redundant combinations, but we still want to be able to combine `authority` or `previous` with specific public keys, so `authority | previous | any | repeated <public_key>` is not enough.

A stricter representation is possible, but a bit more complex, and here the redundancy can't create inconsistency, so it's not too bad imo.